src/proposal.mligo

```diff
@@ -47,12 +46,40 @@ let ensure_proposing_stage(stage_num, store : nat * storage): storage =
   else (failwith("NOT_PROPOSING_STAGE") : storage)
 
 [@inline]
-let ensure_proposal_is_unique (propose_params, store : propose_params * storage): proposal_key =
-  let proposal_key = to_proposal_key(propose_params, Tezos.sender) in
+let ensure_proposal_is_unique (from_, propose_params, store : address * propose_params * storage): proposal_key =
+  let proposal_key = to_proposal_key(propose_params, from_) in
```

Comment:
I think that now that `propose_params` includes the source address we don't need to provide it separately, AFAIU we could change `to_proposal_key` to only have one argument.

Reply:
I've updated this now.